### PR TITLE
No fail checksum

### PR DESCRIPTION
**Pull request by miketypeguy**
This PR handles exceptions on checksum by continuing instead of raising exception. It also fixed a bad key reference in one of the checksum loops

### CODE REVIEW

**Review by ybarot2506 — APPROVED** [review]

**Review by sigJoe — COMMENTED** [review]
Not sure I like this. Shouldn't the change be to do something with the exception rather than ignoring the problem?

**Comment by miketypeguy** [comment]
> Not sure I like this. Shouldn't the change be to do something with the exception rather than ignoring the problem?

Previously it was raising an exception that was not being caught generally by the calling AsyncS3Downloader.  The intended behavior of this was just to inform the user that they requested a sha256 checksum head on an object but that object did not have one associated or could not be retrieved.  It was never mean to stop the operation of the find files function.  The AsyncS3Downloader class in the same module should not fail in this case but rather return the unknown checksum.  Applications like Umpire will use this for a checksum compare which will fail but not inhibit the operation to completion. 

We can re-evaluate this in the future if you want something else.  Right now it doesn't work as intended without this change.

File: maestro/aws/s3.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def find_files(bucket, prefix, case_sensitive=True, connection=None, anonymous=T
                     try:
                         objsum = s3client.head_object(Bucket=bucket, Key=obj.key, ChecksumMode='ENABLED')['ResponseMetadata']['HTTPHeaders']["x-amz-checksum-sha256"]
                     except Exception:
-                        raise ChecksumFailure(f"Unable head object for checksum. Please verify sha256 exists on object {bucket}/{obj.key}")
+                        print(f"Unable head object for checksum. Please verify sha256 exists on object {bucket}/{obj.key}")
```
Comment (haggaret) [inline maestro/aws/s3.py]: nitpick: "unable TO head object..."

File: maestro/aws/s3.py [hunk 2]
```diff
@@ -67,8 +67,7 @@ def find_files(bucket, prefix, case_sensitive=True, connection=None, anonymous=T
                 try:
                     objsum = s3client.head_object(Bucket=bucket, Key=f.key, ChecksumMode='ENABLED')['ResponseMetadata']['HTTPHeaders']["x-amz-checksum-sha256"]
                 except Exception:
-                    raise ChecksumFailure(
-                        f"Unable head object for checksum. Please verify sha256 exists on object {bucket}/{obj.key}")
+                    print(f"Unable head object for checksum. Please verify sha256 exists on object {bucket}/{f.key}")
```
Comment (haggaret) [inline maestro/aws/s3.py]: Same deal here

**Review by haggaret — APPROVED** [review]

**Review by ryan-j-mci — APPROVED** [review]

File: maestro/aws/s3.py (hunk 1 above)
Comment (miketypeguy) [reply maestro/aws/s3.py]: I will fix that 👍

**Pull request merged**